### DISCUSSION TITLE
Disscussion: Is 'table metadata not loaded yet' reasonable?

### DISCUSSION THREAD

**Discussion by ZENOTME** [icelake-io/icelake#80] [Ideas]
I find that the `current_table_metadata` interface will check `self.current_version` and return "table metadata not loaded yet" when it's 0.
```
    pub fn current_table_metadata(&self) -> Result<&types::TableMetadata> {
        if self.current_version == 0 {
            return Err(anyhow!("table metadata not loaded yet"));
        }
        ...
    }
```
But I find that it will call `load` automatically in the `open` or `open_with_op`. So I guesss the error will occur in this case: `self.current_version = metadata.last_updated_ms; // metadata.last_updated_ms = 0 `
And we hope user to fix this error like this:
```
let table = Table::open("");
table.current_table_metadata(); // get error
table.load(); // retry load 
```

But why not directly return error in `open` when we can't load a valid current version. And we can assert current_version is valid in `current_table_metadata`. Like 
```
let table = Table::open(""); // Will return error if the metadata.last_updated_ms = 0 

pub fn current_table_metadata(&self) -> Result<&types::TableMetadata> {
    assert(self.current_version != 0)
    ...
}
```
I think this way is more friendly for user🤔 When it open succeesfully, it can convince that the version is valid. It don't need to check and process error every time in later operation.

**Comment by ZENOTME** [comment]
How do you think? @Xuanwo @liurenjie1024

**Comment by Xuanwo** [comment]
Makes sense, let's do this.

**Comment by liurenjie1024** [comment]
Yeah, I think returning error in open makes more sense.
